client/src/lib.rs

```diff
@@ -0,0 +1,143 @@
+//! Client libarary for the job runner
+
+use std::convert::TryInto;
+
+use tokio::sync::mpsc::Sender;
+use tokio::time::{sleep, Duration};
+use tonic::transport::{Certificate, Channel, ClientTlsConfig, Identity};
+
+use common::t_service_client::TServiceClient;
+use common::*;
+
+// Re-exports
+pub use common::output_event::Stream as OutputStream;
+pub use common::{JobId, JobStartRequest};
+
+#[derive(Debug, Clone)]
+pub struct TlsConfig {
+    /// Contents of the PEM-formatted server root CA certificate file
+    pub server_root_ca_crt: Vec<u8>,
+    /// Contents of the PEM-formatted client certificate file
+    pub client_crt: Vec<u8>,
+    /// Contents of the PEM-formatted client private key file
+    pub client_key: Vec<u8>,
+}
+
+// TODO: create a proper error type, and add conversions to it
+
+type DResult<T> = Result<T, Box<dyn std::error::Error>>;
+
+pub struct Client {
+    client: TServiceClient<Channel>,
+}
+impl Client {
+    /// Connect to a job server
+    pub async fn connect(target: String, tls: TlsConfig) -> DResult<Self> {
+        let server_root_ca_crt = Certificate::from_pem(tls.server_root_ca_crt);
+        let client_identity = Identity::from_pem(tls.client_crt, tls.client_key);
+
+        let tls = ClientTlsConfig::new()
+            .domain_name("localhost")
+            .ca_certificate(server_root_ca_crt)
+            .identity(client_identity);
+
+        log::debug!("Connecting to {}", target);
+        let channel = Channel::from_shared(target)?
+            .tls_config(tls)?
+            .connect()
+            .await?;
+        log::debug!("Connected");
+
+        Ok(Self {
+            client: TServiceClient::new(channel),
+        })
+    }
+
+    /// Starts a new job by spawning a process from given executable path and arguments.
+    /// Returns a unique job id (UUID v4), that is used to specify the target job for other endpoints.
+    // If the executable is not found or cannot be executed, immediately returns an error.
+    pub async fn start(&mut self, req: JobStartRequest) -> DResult<JobId> {
+        let command = tonic::Request::new(req);
+        let response = self.client.start(command).await?;
+        Ok(response
+            .into_inner()
+            .try_into()
+            .expect("Server returned invalid JobId"))
+    }
+
+    /// Cancels a job by sending `SIGKILL`.
+    /// This is done asynchronously, and can return before the process has terminated.
+    /// If you must wait until the job has stopped, it can do so by calling `wait`.
```

Comment:
Calling sigkill on a stopped job makes no sense.

Reply:
Clarified in d5b1e53.